datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx

```diff
@@ -10,6 +11,25 @@ import { Preview } from './preview/Preview';
 export class DatasetEntity implements Entity<Dataset> {
     type: EntityType = EntityType.Dataset;
 
+    icon = (fontSize: number, styleType: IconStyleType) => {
+        if (styleType === IconStyleType.TAB_VIEW) {
+            return <DatabaseOutlined style={{ fontSize }} />;
+        }
+
+        if (styleType === IconStyleType.HIGHLIGHT) {
+            return <DatabaseFilled style={{ fontSize, color: '#B37FEB' }} />;
+        }
+
+        return (
+            <DatabaseFilled
+                style={{
+                    fontSize,
+                    color: '#BFBFBF',
+                }}
+            />
+        );
```

Comment:
same comment here w/ making the caller of `icon()` provide the color

Reply:
The caller shouldn't know this detail. Otherwise it makes it difficult to iterate a list of entities and get their icons.